plugin/trino-hudi/pom.xml

```diff
@@ -15,7 +15,7 @@
 
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
-        <dep.hudi.version>0.12.2</dep.hudi.version>
+        <dep.hudi.version>0.12.3</dep.hudi.version>
```

Comment:
Could you send PR for https://github.com/trinodb/docker-images/blob/master/testing/spark3-hudi/Dockerfile? 

Also, please update `hudi.rst`.

Reply:
I started the release process of new docker-images https://github.com/trinodb/docker-images/actions/runs/4946599101. 

Please update the image version and `hudi.rst`.